### PR TITLE
Fix: convert books to functional component

### PR DESCRIPTION
**Pull request by akankshat05**
Fixes : [T348414](https://phabricator.wikimedia.org/T348414)
## Proposed Changes
- Books into a functional component and used the useState hook
## Files Updated
- `books.js`

### CODE REVIEW

File: components/Books.js [hunk 1]
```diff
@@ -1,52 +1,31 @@
...
+    setIAIdentifier("");
+    setInputDisabled(false);
   };
 
-  /**
-   * Change the example when user selects a different library
-   * @return {String}
-   */
-  showExample = () => {
+  const showExample = () => {
```
Comment (Parthiv-M) [inline components/Books.js:26]: Looks like this function is unused and can be removed

File: components/Books.js [hunk 2]
```diff
@@ -60,15 +39,14 @@ class Books extends React.Component {
...
     }
     return url;
   };
 
-  /**
-   * Makes a request from Google Books API based on the entered book Id
-   * If the book Id is not valid the request resolves with an error message
-   */
-  validateGoogleBook = (enteredId) => {
+  const validateGoogleBook = (enteredId) => {
```
Comment (Parthiv-M) [inline components/Books.js:49]: Looks like this function is unused and can be removed

**Comment by Parthiv-M** [comment]
At some point, we should remove the `withSession` hook as it will remain unused. If #191 gets merged before this PR, then `withSession` can be removed here itself.

File: components/Books.js (hunk 2 above)
Comment (akankshat05) [reply components/Books.js]: I have removed the unused functions.

**Comment by coderwassananmol** [comment]
@akankshat05 Please remove the `withSession` hook and resolve the conflicts.

**Review by Parthiv-M — APPROVED** [review]
LGTM

**Pull request merged**